GVFS/GVFS.UnitTests.Windows/Windows/Upgrader/NewProductUpgradeTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.NuGetUpgrade;
+using GVFS.Tests.Should;
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Windows.Mock.Upgrader;
+using Moq;
+using NUnit.Framework;
+
+namespace GVFS.UnitTests.Windows.Windows.Upgrader
+{
+    public class NewProductUpgradeTests

```

Comment:
I don't like the name, but there is an existing test class for `ProductUpgraderTests` whose testing style I did not want to mix with these tests, as they have different setups.

Reply:
`TryCreateUpgraderTests`? or will you be adding more tests to this later?